renderer/components/Calendar/Calendar.tsx

```diff
@@ -139,13 +152,96 @@ export function Calendar({
           isValid: activities.every((report: ReportActivity) => report.isValid === true),
         };
       });
-      setMonthWorkHoursReports(monthWorkHours);
+
+      setFormattedQuarterReports(fromattedReports);
     } catch (err) {
       console.log("Error details ", err)
       setRenderError({errorTitle:"Calendar error", errorMessage:"An error occurred when validating reports for the last month. "})
+    }
+  }, [parsedQuarterReports, daysOff]);
 
-}
-  }, [monthReportsFromServer]);
+  const loadHolidaysAndVacations = async () => {
+    try {
+      const timetrackerUserInfo = JSON.parse(
+        localStorage.getItem("timetracker-user")
+      );
+      const plannerToken = timetrackerUserInfo?.accessToken;
+      const userPromises = [];
+
+      const holidaysPromise = global.ipcRenderer.invoke(
+        "timetracker:get-holidays",
+        plannerToken
+      );
+
+      const userEmail = timetrackerUserInfo?.email;
+      const vacationsPromise = global.ipcRenderer.invoke(
+        "timetracker:get-vacations",
+        plannerToken,
+        userEmail
+      );
+
+      userPromises.push(...[holidaysPromise, vacationsPromise]);
```

Comment:
if i'm not mistaken you can write it like this `userPromises.push(holidaysPromise, vacationsPromise);`